indexing-service/src/main/java/org/apache/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -540,7 +540,9 @@ public boolean isWorkerRunningTask(ZkWorker worker, String taskId)
     } else if ((completeTask = completeTasks.get(task.getId())) != null) {
       return completeTask.getResult();
     } else {
-      return addPendingTask(task).getResult();
+      RemoteTaskRunnerWorkItem workItem = addPendingTask(task);
+      runPendingTasks();
```

Comment:
I find it confusing and error prone that after adding a task(s) now you have to explicitly call run..., maybe ad a comment to make it clear that it is required (else?)